Sources/ZIPFoundation/FileManager+ZIP.swift

```diff
@@ -124,8 +124,10 @@ extension FileManager {
         }
     }
 
-    class func attributes(from centralDirectoryStructure: CentralDirectoryStructure) -> [FileAttributeKey: Any] {
-        var attributes = [.posixPermissions: defaultPermissions,
+    class func attributes(from centralDirectoryStructure: CentralDirectoryStructure,
```

Comment:
This would be a good opportunity to refactor this method. We could rename it to `attributes(for entry: Entry)` and retrieve CDS, `entryType` and `osType` from the passed in `entry` struct.